pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
 		return docker.AuthConfiguration{}, false
 	}
 
-	var cfg credentialprovider.DockerConfig
```

Comment:
Since you are cleaning up code in this area anyway, i'm pretty sure the else block on lines 44-46 are unnecessary, no?  GetDockercfgFile already handles the case where pathForAuthType might be an empty string.